verl/utils/dataset/sampler.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2025 Amazon.com Inc and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import abstractmethod
+from collections.abc import Sized
+
+from omegaconf import DictConfig
+from torch.utils.data import Sampler
+
+from verl import DataProto
+
+
+class AbstractSampler(Sampler[int]):
+    @abstractmethod
+    def __init__(
+        self,
+        data_source: Sized,
+        config: DictConfig,
+    ):
+        pass
+
+
+class AbstractCurriculumSampler(AbstractSampler):
+    @abstractmethod
```

Comment:
for curriculum learning, my gut is that it's quite hard to get it right at this point since the algorithm can be quite diverse and it changes quickly. the update API seems too general and I suspect it may get changed in the future. Shall we rename it to be a less ambiguous name such as `on_batch_end`? I also want to move this to `verl/experimental` since the API is likely to evolve overtime @frrad